state/constraints.go

```diff
@@ -114,3 +114,21 @@ func writeConstraints(mb modelBackend, id string, cons constraints.Value) error
 	}
 	return nil
 }
+
+// ConstraintsBySpaceName returns all the constraints given by the spaceName, as a map keyed by the DocID.
+func (st *State) ConstraintsBySpaceName(spaceName string) (map[string]constraints.Value, error) {
```

Comment:
There is actually no need to have both this method and `GetConstraintsOps`.

Just have one method `ConstraintsOpsForSpaceNameChange`, find the docs and return an operation for each one. Then you don't need to leak out maps with DocIDs from state.

Reply:
I intentionally wrote the logic outside in the `rename.go` file. 

To comply with your review I have 2 ideas: 
- move the space logic into the `ConstraintsOpsForSpaceNameChange` 
- give the `ConstraintsOpsForSpaceNameChange` a kind `filter func()` as a second param to apply the space changes. 

Or did you have something else in mind?

Reply:
The purpose of using the `ModelOperation` is to eliminate cross-dependencies between different state modules just to get transactional execution. In this case spaces/constraints/settings.

Given that both the identification and the edit ops for constraints and a concern only of constraints, there is no need to double handle in the model operation.

So the proposed method would be:
```go
func (st *State) ConstraintsOpsForSpaceNameChange(from, to string) ([]txn.Op, err)
```
Do the query that you are doing in `ConstraintsBySpaceName` and for each, accrue and return the ops as `GetConstraintsOps` does. By the way, you don't need to validate the constraints - we have valid constraints that we're changing with a valid space name.

Reply:
Okay, I moved the logic out of the package to state/constraint :+1: